src/show/pages/public/SignUp/index.js

```diff
@@ -1,9 +1,157 @@
+  });
+
+  const isFormValid = Object.values(validations).every(Boolean) && inputs.email && inputs.password && inputs.region && inputs.invitationCode && inputs.accepted;
 
-function SignUp() {
   return (
-    <div>SignUp</div>
-  )
-}
+    <Split 
+      leftContent={<img src={LeftImage} alt="Left side" className="object-cover h-full w-full" />}
+      rightContent={
+        <div className="flex justify-center my-5">
+          <div className="w-64 bg-white">
+            <h2 className="text-2xl font-bold mb-4">Create your account</h2>
+            <form onSubmit={handleSubmit} className="signup-form">
+              <TextField
+                label="Company name"
+                type="text"
+                name="Companyname"
+                placeholder="Tesla"
+                value={inputs.Companyname}
+                onChange={(value) => handleInputChange('Companyname', value, setInputs, setValidations)}
+                required
+              />
+              <TextField
+                label="Email"
+                type="email"
+                name="email"
+                placeholder="Enter Email"
+                value={inputs.email}
+                onChange={(value) => handleInputChange('email', value, setInputs, setValidations)}
+                required
+              />
+              <Region
+                inputs={inputs}
+                handleInputChange={handleInputChange}
+                setInputs={setInputs}
+              />
+              <TextField
+                label="Password"
+                type="password"
+                name="password"
+                placeholder="Enter Password"
+                value={inputs.password}
+                onChange={(value) => handleInputChange('password', value, setInputs, setValidations)}
+                required
+              />
+              <div className="mt-2">
+                {inputs.password && (
+                  <>
+                    <p className={`flex textclr ${validations.length ? 'text-green-500' : 'text-red-500'}`}>
+                      {validations.length ? (
+                        <CheckIcon className="h-4 w-4 mr-2" />
+                      ) : (
+                        <XMarkIcon className="h-4 w-4 mr-2" />
+                      )}
+                      {validations.length ? 'At least 8 characters' : 'At least 8 characters required'}
+                    </p>
+                    <p className={`flex textclr ${validations.number ? 'text-green-500' : 'text-red-500'}`}>
+                      {validations.number ? (
+                        <CheckIcon className="h-4 w-4 mr-2" />
+                      ) : (
+                        <XMarkIcon className="h-4 w-4 mr-2" />
+                      )}
+                      {validations.number ? 'Contains a number' : 'Number required'}
+                    </p>
+                    <p className={`flex textclr ${validations.specialCharacter ? 'text-green-500' : 'text-red-500'}`}>
```

Comment:
```
<div className="mt-2">
                {inputs.password && (
                  <>
                    <p className={`flex textclr ${validations.length ? 'text-green-500' : 'text-red-500'}`}>
                      {validations.length ? (
                        <CheckIcon className="h-4 w-4 mr-2" />
                      ) : (
                        <XMarkIcon className="h-4 w-4 mr-2" />
                      )}
                      {validations.length ? 'At least 8 characters' : 'At least 8 characters required'}
                    </p>
                    <p className={`flex textclr ${validations.number ? 'text-green-500' : 'text-red-500'}`}>
                      {validations.number ? (
                        <CheckIcon className="h-4 w-4 mr-2" />
                      ) : (
                        <XMarkIcon className="h-4 w-4 mr-2" />
                      )}
                      {validations.number ? 'Contains a number' : 'Number required'}
                    </p>
                    <p className={`flex textclr ${validations.specialCharacter ? 'text-green-500' : 'text-red-500'}`}>
                      {validations.specialCharacter ? (
                        <CheckIcon className="h-4 w-4 mr-2" />
                      ) : (
                        <XMarkIcon className="h-4 w-4 mr-2" />
                      )}
                      {validations.specialCharacter ? 'Contains a special character' : 'Special character required'}
                    </p>
                  </>
                )}
              </div>
```

this whole thing is repeated, which doesn't look good, create an array and if any one condition set a key to be true in there

Reply:
done